### PR TITLE
[OASIS-71] Fix datadog exporter configs

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

File: comp/otelcol/otlp/components/exporter/serializerexporter/go.mod [hunk 1]
```diff
@@ -83,6 +83,8 @@ require (
 	go.uber.org/multierr v1.11.0
 )
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/exporter/serializerexporter/go.mod:85]: move to require block below

File: cmd/otel-agent/config/agent_config.go [hunk 2]
```diff
@@ -69,8 +69,14 @@ func NewConfigComponent(ctx context.Context, uris []string) (config.Component, e
 	pkgconfig.Set("api_key", apiKey, pkgconfigmodel.SourceLocalConfigProcess)
 	pkgconfig.Set("site", site, pkgconfigmodel.SourceLocalConfigProcess)
 
+	pkgconfig.Set("dd_url", ddc.Metrics.Endpoint, pkgconfigmodel.SourceLocalConfigProcess)
+
+	// Log configs
 	pkgconfig.Set("logs_enabled", true, pkgconfigmodel.SourceLocalConfigProcess)
-	pkgconfig.Set("logs_config.use_compression", true, pkgconfigmodel.SourceLocalConfigProcess)
+	pkgconfig.Set("logs_config.logs_dd_url", ddc.Logs.Endpoint, pkgconfigmodel.SourceLocalConfigProcess)
+	pkgconfig.Set("logs_config.batch_wait", ddc.Logs.BatchWait, pkgconfigmodel.SourceLocalConfigProcess)
```
Comment (dineshg13) [inline cmd/otel-agent/config/agent_config.go:77]: can u add tests for this ?

File: comp/otelcol/otlp/components/exporter/serializerexporter/go.mod (hunk 1 above)
Comment (songy23) [reply comp/otelcol/otlp/components/exporter/serializerexporter/go.mod]: done

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**